modules/bootconf.py

```diff
@@ -118,3 +127,94 @@ def configure(self, old, new):
             return False
 
         return True
+
+class UBootBeagleboneBootloader(BootloaderConfigurator):
+
+    def switchUEnv(self, old, new, uEnvPath):
+        ''' Switch boot partition using uEnv bootpart variable '''
+        oldidx = getPartitionIndex(old)
+        newidx = getPartitionIndex(new)
+        if not oldidx or not newidx:
+            return False
+
+        if not super(UBootBeagleboneBootloader, self).applyTextTransformation(uEnvPath, 'bootpart=1:' + oldidx, 'bootpart=1:' + newidx):
+            return False
+
+        return True
+
+    def tweakUEnv(self, bootmountpoint):
+        ''' Various tweaks and cleanups to uEnv.txt '''
+
+        log.info ("tweakUEnv: Tweaking and cleaning up uEnv.txt...")
+
+        if not isMounted(bootmountpoint):
+            log.error("transformUEnv: " + bootmountpoint + " is not a mountpoint.")
+            return False
+        uEnvPath = os.path.join(bootmountpoint, 'uEnv.txt')
+        if not os.path.isfile(uEnvPath):
+            log.error("transformUEnv: uEnv.txt seem not to exist in boot partition")
+            return False
+
+        lines=[]
+        fixFindUUID = True
+        finduuid = 'finduuid=part uuid mmc ${bootpart} uuid\n'
+        with open(uEnvPath) as f:
+            for line in f:
+                # Remove setemmcroot lines as current uboot is not using it anymore
+                if "setemmcroot" in line:
+                    continue
+                # Tweak finduuid to use a configurable partition
+                if line.startswith('finduuid='):
+                    line = finduuid
+                    fixFindUUID = False
+
+                lines.append(line)
+            # If no finduuid found append one
+            if fixFindUUID:
+                line.append('finduuid=part uuid mmc ${bootpart} uuid\n')
+        with open(uEnvPath + ".tmp", "w") as f:
+            for line in lines:
+                f.write(line)
+            # Dump to disk - no need to fsync the parent directory
+            # As we are editing a file - not creating one
+            os.fsync(f)
+        os.rename(uEnvPath + ".tmp", uEnvPath)
```

Comment:
An `fsync()` on the directory fd is needed after rename to force the rename to be flushed to disk